ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -324,6 +324,15 @@ public TransientQueryMetadata executeTransientQuery(
     }
   }
 
+  public KsqlConfig getKsqlConfig() {
```

Comment:
this needs to be synchronized with `alterSystemProperty`

Reply:
With synchronization you mean using the `ksqlConfig` of the `EngineContext` when doing this: `this.ksqlConfig = this.ksqlConfig.cloneWithPropertyOverwrite(overrides);`, right?

Reply:
I mean synchronzation from a concurrency perspective. The ksql config object (both here and in primaryContext) is not `volatile`, so you might read the value while it's being written and get back a corrupted value. To fix this you can either make it volatile, or make `alterSystemProperty` and `getKsqlConfig` synchronized.